core/src/main/scala/org/apache/spark/scheduler/EventLoggingListener.scala

```diff
@@ -86,8 +86,12 @@ private[spark] class EventLoggingListener(
    * Creates the log file in the configured log directory.
    */
   def start() {
-    if (!fileSystem.isDirectory(new Path(logBaseDir))) {
-      throw new IllegalArgumentException(s"Log directory $logBaseDir does not exist.")
+    try {
+      if (!fileSystem.getFileStatus(new Path(logBaseDir)).isDir()) {
+        throw new IllegalArgumentException(s"Log directory $logBaseDir is not a directory.")
+      }
+    } catch {
+      case ioe: FileNotFoundException => fileSystem.mkdirs(new Path(logBaseDir))
```

Comment:
Per @pwendell's earlier comment, I think we should check whether `logBaseDir` already exists and log a warning if it's being automatically created to let users know that it could potentially have wider directory permissions than they intended.